modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -103,6 +135,17 @@ private boolean doBinaryComparisonCoercion(SqlCallBinding binding) {
         RelDataType leftType = binding.getOperandType(0);
         RelDataType rightType = binding.getOperandType(1);
 
+        if (SqlTypeUtil.isIntType(leftType) && SqlTypeUtil.isCharacter(rightType)
```

Comment:
Guards for string-numeric literals are only added for int-s. Comparison with floats and doubles is still going to produce exceptions.

Reply:
can you explain what kind of tests i forgot to append in such a case plz ?